plugins/jobs/src/js/JobStopRun.js

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import { componentFromStream } from "data-service";
+import { stopJobRun } from "#SRC/js/events/MetronomeClient";
+
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import JobStopRunModal from "./components/JobStopRunModal";
+
+function executeStop({ jobID, jobRun, onSuccess }) {
+  return stopJobRun(jobID, jobRun)
+    .map(_ => ({ done: true }))
+    .do(_ => onSuccess())
+    .startWith({ done: false });
+}
+
+function stopOperation() {
+  const stopSubject$ = new Subject();
+  const stop$ = stopSubject$.switchMap(executeStop).catch(error => {
+    return stop$.startWith({
+      errorMsg: error.response.message,
+      done: true
+    });
+  });
+
+  return {
+    stop$,
+    stopHandler: (jobID, jobRun, onSuccess) => {
+      stopSubject$.next({ jobID, jobRun, onSuccess });
+    }
+  };
+}
+
+const JobStopRun = componentFromStream(props$ => {
+  const { stop$, stopHandler } = stopOperation();
+  const stopEmit$ = stop$.startWith({ done: null });
```

Comment:
Why do we default `executeStop` to `{done: false}` just to redefine it here to `{done: null}`?

Reply:
If we could generalize `startWith` we wouldn't need coming up we a name like `stopEmit$` if we can't I'd be looking for a better name.

Reply:
There is a semantic difference from null and false. `null` means I never made a request, `false` means I am in the middle of one.  Because we combine latest we need to emit once, so the component renders. 

Notice that `executeStop` is inside `switchMap` so it is not going to be triggered at loading. So what actually happens is:

We load the component with `done: null`, when we trigger a request done alternate between `false` and `true`, staying at `true` most of the time when a request is not happening.